Possible exception during validation of IPAddressField

I faced with not expected behaviour of IPAddress field during validation certain types of input.

## Checklist

- [x] I have verified that that issue exists against the `master` branch of Django REST framework.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](http://www.django-rest-framework.org/topics/third-party-resources/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [x] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

## Steps to reproduce

1. Create a serializer with IPAddressField
2. Pass not iterable object as input for the serializer. E.g. integer.

## Expected behavior

Validation error with usual text:
```['Enter a valid IPv4 or IPv6 address.']```

## Actual behavior
Exception:

```
rest_framework/fields.py:701: in run_validation
    return super(CharField, self).run_validation(data)
rest_framework/fields.py:487: in run_validation
    value = self.to_internal_value(data)
rest_framework/fields.py:807: in to_internal_value
    if data and ':' in data:
E   TypeError: argument of type 'int' is not iterable
```

--- a/rest_framework/fields.py
+++ b/rest_framework/fields.py
@@ -804,7 +804,10 @@ class IPAddressField(CharField):
         self.validators.extend(validators)
 
     def to_internal_value(self, data):
-        if data and ':' in data:
+        if not isinstance(data, six.string_types):
+            self.fail('invalid', value=data)
+
+        if ':' in data:
             try:
                 if self.protocol in ('both', 'ipv6'):
                     return clean_ipv6_address(data, self.unpack_ipv4)

--- a/tests/test_fields.py
+++ b/tests/test_fields.py
@@ -663,6 +663,7 @@ class TestIPAddressField(FieldValues):
         '127.122.111.2231': ['Enter a valid IPv4 or IPv6 address.'],
         '2001:::9652': ['Enter a valid IPv4 or IPv6 address.'],
         '2001:0db8:85a3:0042:1000:8a2e:0370:73341': ['Enter a valid IPv4 or IPv6 address.'],
+        1000: ['Enter a valid IPv4 or IPv6 address.'],
     }
     outputs = {}
     field = serializers.IPAddressField()

